File: Python/Algorithm/DEC_to_HEX.py
# Complexity: O(n)
# ASCII: 48 = '0', 55 = 'A', 97 = 'a'

def hex(n):
    if n == 0: return "0"
    s = ""
    while n:
        tmp = n % 16
        if tmp < 10: s += chr(tmp + 48)  # Adding 48 to convert digit to ASCII
        else: s += chr(tmp + 55)  # Adding 55 to convert 10-15 to 'A'-'F'
        n >>= 4
    return s[::-1]  # Reverse the string

File: CPP/Algorithm/DEC_to_HEX.cpp
// Complexity: O(n)
// ASCII: 48 = '0', 55 = 'A', 97 = 'a'
// Built-in: cout << hex << n;  // In Lowercase

auto hex(auto n) {
    string s;
    while(n) {
        int tmp = n % 16;
        s += (tmp + (tmp < 10 ? 48 : 55));
        n >>= 4;
    }
    reverse(s.begin(), s.end()); return s;
}